src/main/java/org/elasticsearch/index/translog/fs/FsTranslog.java

```diff
@@ -162,20 +205,25 @@ public void close() throws IOException {
 
     @Override
     public long currentId() {
-        FsTranslogFile current1 = this.current;
-        if (current1 == null) {
-            return -1;
+        try (ReleasableLock _ = readLock.acquire()) {
+            return current.translogId();
         }
-        return current1.id();
     }
 
     @Override
     public int estimatedNumberOfOperations() {
-        FsTranslogFile current1 = this.current;
-        if (current1 == null) {
-            return 0;
+        int ops = 0;
+        try (ReleasableLock _ = readLock.acquire()) {
```

Comment:
Same here, maybe add that this has a cost, so we need to protect logging statements that use `estimatedNumberOfOperations` so it doesn't cause additional locking when verbose logging is disabled